### PR TITLE
Refactor `CRaceHelper`, mark `CCollision` functions as `const` and add tile getters

### PR DESCRIPTION
**Pull request by Robyt3**
See commit messages.

## Checklist

- [X] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Review by def- — APPROVED** [review]

File: src/game/client/race.cpp [hunk 1]
```diff
@@ -77,19 +77,15 @@ bool CRaceHelper::IsStart(CGameClient *pClient, vec2 Prev, vec2 Pos)
...
-				if(pCollision->GetTileIndex(Indice) == TILE_START)
-					return true;
-				if(pCollision->GetFTileIndex(Indice) == TILE_START)
-					return true;
-			}
-		else
+		if(vIndices.empty())
 		{
-			if(pCollision->GetTileIndex(pCollision->GetPureMapIndex(Pos)) == TILE_START)
+			vIndices.push_back(pCollision->GetPureMapIndex(Pos));
```
Comment (heinrich5991) [inline src/game/client/race.cpp]: This introduces an allocation in the common case, which sounds unnecessary. Perhaps we could construct an iterator without pushing it into the vector?

File: src/game/client/race.h [hunk 2]
```diff
@@ -3,17 +3,23 @@
...
-	static int ms_aFlagIndex[2];
+	void Init(const CGameClient *pGameClient);
 
 	// these functions return the time in milliseconds, time -1 is invalid
-	static int TimeFromSecondsStr(const char *pStr); // x.xxx
-	static int TimeFromStr(const char *pStr); // x minute(s) x.xxx second(s)
-	static int TimeFromFinishMessage(const char *pStr, char *pNameBuf, int NameBufSize); // xxx finished in: x minute(s) x.xxx second(s)
+	int TimeFromSecondsStr(const char *pStr) const; // x.xxx
+	int TimeFromStr(const char *pStr) const; // x minute(s) x.xxx second(s)
+	int TimeFromFinishMessage(const char *pStr, char *pNameBuf, int NameBufSize) const; // xxx finished in: x minute(s) x.xxx second(s)
```
Comment (heinrich5991) [inline src/game/client/race.h:20]: What's the motivation for making these three functions non-static?

**Review by heinrich5991 — COMMENTED** [review]
Found an unneeded allocation, looks good to me otherwise.

File: src/game/client/race.cpp (hunk 1 above)
Comment (Robyt3) [reply src/game/client/race.cpp]: I didn't like the readability of the resulting code with iterators, so I reverted to the previous structure again (with minor refactoring). Maybe C++20 could make it cleaner eventually.

File: src/game/client/race.h (hunk 2 above)
Comment (Robyt3) [reply src/game/client/race.h]: I guess for consistency with the other functions of this class, but it seems like an unnecessary change so I reverted it.

**Review by heinrich5991 — APPROVED** [review]
Thanks!

**Pull request merged**